jobManager.py

```diff
@@ -44,32 +44,48 @@ def run(self):
 
     def __manage(self):
         self.running = True
+        # job-associated instance id
+        nextId = 10000
         while True:
-            if Config.REUSE_VMS:
-                id, vm = self.jobQueue.getNextPendingJobReuse()
-            else:
-                id = self.jobQueue.getNextPendingJob()
+            id = self.jobQueue.getNextPendingJob()
 
             if id:
                 job = self.jobQueue.get(id)
+                if not job.accessKey and Config.REUSE_VMS:
+                    id, vm = self.jobQueue.getNextPendingJobReuse(id)
+                    job = self.jobQueue.get(id)
+
                 try:
                     # Mark the job assigned
                     self.jobQueue.assignJob(job.id)
-
-                    # Try to find a vm on the free list and allocate it to
-                    # the worker if successful.
-                    if Config.REUSE_VMS:
-                        preVM = vm
+                    # if the job has specified an account
+                    # create an VM on the account and run on that instance
+                    if job.accessKeyId:
+                        from vmms.ec2SSH import Ec2SSH
+                        vmms = Ec2SSH(job.accessKeyId, job.accessKey)
+                        newVM = copy.deepcopy(job.vm)
+                        newVM.id = nextId
+                        preVM = vmms.initializeVM(newVM)
+                        if nextId > 99999:
+                            nextId = 1000
```

Comment:
No. It is not job id but an intern id, a unique identifier, for VMs. It could also be generate via random(), but I use this in an attempt to be consistent with the original code, which uses hardcoded number between 1000 and 9999. To be more specific, I make a small change.